### PR TITLE
Add command history navigation to CommandBox

### PR DESCRIPTION
**Pull request by winstonzhaozhekai**
Implement terminal-style command history navigation using arrow keys

Updated CommandBox.java to support UP/DOWN arrow navigation

Added logic reference and history tracking fields

Implemented methods for navigating through command history

Preserved current user input when entering/exiting history mode

Reset navigation state after command execution

Modified MainWindow.java to pass Logic reference to CommandBox

Added CommandBoxTest.java with 23 comprehensive test cases covering:

Command history storage, navigation, and edge cases

Equality, immutability, and copy constructor behavior in CommandHistory

This feature enhances usability by allowing users to quickly recall and reuse previously executed commands, improving workflow efficiency and aligning the application with familiar terminal interaction patterns.

### CODE REVIEW

File: src/main/java/seedu/coursebook/ui/CommandBox.java [hunk 1]
```diff
@@ -44,11 +68,69 @@ private void handleCommandEntered() {
...
+    private void showPreviousCommand() {
+        ObservableList<String> history = logic.getHistory();
+        if (history.isEmpty()) {
+            return;
+        }
+
+        // If not currently in history, save current text and start from most recent command
+        if (historyIndex == -1) {
+            currentEditText = commandTextField.getText();
+            historyIndex = history.size() - 1;
```
Comment (subhitsha-sp) [inline src/main/java/seedu/coursebook/ui/CommandBox.java]: Thoughtful handling of edge cases (e.g., empty history, restoring original input)
